app/controllers/tags_controller.rb

```diff
@@ -16,14 +16,16 @@ def show
       end
 
       format.rss do
-        @statuses = Status.as_tag_timeline(@tag).limit(PAGE_SIZE)
+        @statuses = HashtagQueryService.new.call(@tag, params.slice(:any, :all, :none)).distinct.limit(PAGE_SIZE)
```

Comment:
If `distinct` is called on all uses of the service, why not put it inside the service?

Reply:
Ah, it was because rails 5's `or` method is verrrry picky about what kinds of clauses you can join via or (and distinct was blowing it up), but I believe we can pull it into the service by calling `distinct` on both the original query and the 'or' one.